### ISSUE TITLE
Support Error.prepareStackTrace and Error.captureStackTrace

### ISSUE THREAD

**Issue by jianchun** [Microsoft/ChakraCore#587]
ChakraCore supports Error.stackTraceLimit, but not the other [StackTrace APIs in V8](https://github.com/v8/v8/wiki/Stack%20Trace%20API). Found multiple sites and nodejs modules using other APIs and incompatible with Edge / node-chakracore.

To improve compatibility, we need to consider better support of other APIs, Error.prepareStackTrace and Error.captureStackTrace.

Error.prepareStackTrace may be a bit harder to implement as we need to expose current native stack trace frame data structures to JS.

**Comment by jianchun** [comment]
@abchatra

**Comment by abchatra** [comment]
This [website](https://invest.shareview.co.uk/sview/trade/certificated/trade.htm) is dependent on Error.captureStackTrace. See OS bug [6767443](https://microsoft.visualstudio.com/DefaultCollection/OS/_workitems?_a=edit&id=6767443) for details

**Comment by bterlson** [comment]
@RichiCoder1 can you expand?

**Comment by RichiCoder1** [comment]
I'm not sure how well I can explain, but this issue relates:
https://github.com/petkaantonov/bluebird/issues/1274

TL;DR appears to be that Bluebird tries to use Error.captureStackTrace, and failing that (new Error()).stack. Apparently the most recent version of edge doesn't populate `stack` when Error isn't throw, so Bluebird simply fails.
